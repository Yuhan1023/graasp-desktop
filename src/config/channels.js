// this file needs to use module.exports as it is used both by react and electron
module.exports = {
  SAVE_SPACE_CHANNEL: 'space:save',
  GET_SPACE_CHANNEL: 'space:get',
  GET_SPACES_CHANNEL: 'spaces:get',
  DELETE_SPACE_CHANNEL: 'space:delete',
  DELETED_SPACE_CHANNEL: 'space:deleted',
  LOAD_SPACE_CHANNEL: 'space:load',
  LOADED_SPACE_CHANNEL: 'space:loaded',
  EXPORT_SPACE_CHANNEL: 'space:export',
  EXPORTED_SPACE_CHANNEL: 'space:exported',
  SHOW_OPEN_DIALOG_CHANNEL: 'show-open-dialog',
  SHOW_SAVE_DIALOG_CHANNEL: 'show-save-dialog',
  OPEN_DIALOG_PATHS_SELECTED_CHANNEL: 'open-dialog-paths-selected',
  SHOW_MESSAGE_DIALOG_CHANNEL: 'show-message-dialog',
  SAVE_DIALOG_PATH_SELECTED_CHANNEL: 'save-dialog-path-selected',
  MESSAGE_DIALOG_RESPOND_CHANNEL: 'message-dialog-respond',
};
