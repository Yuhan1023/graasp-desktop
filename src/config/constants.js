// phase item types
export const TEXT = 'text/html';
export const VIDEO = new RegExp('video/*');
export const IMAGE = new RegExp('image/*');
export const RESOURCE = 'Resource';
export const APPLICATION = 'Application';
export const IFRAME = 'application/octet-stream';
export const DEFAULT_RADIUS = 50;
export const DEFAULT_LANGUAGE = 'en';
export const DEFAULT_DEVELOPER_MODE = false;
export const DEFAULT_GEOLOCATION_ENABLED = false;
export const SHORT_ID_LENGTH = 6;
export const LONG_ID_LENGTH = 24;
export const SMART_GATEWAY_HOST = 'gateway.golabz.eu';
export const SMART_GATEWAY_QUERY_STRING_DIVIDER = '#';
export const CONTROL_TYPES = {
  BUTTON: 'BUTTON',
  SWITCH: 'SWITCH',
};
export const MIN_CARD_WIDTH = 345;
export const DEFAULT_PROTOCOL = 'https';

// math
export const BLOCK_MATH_DIV = 'p';
export const INLINE_MATH_DIV = 'span';
export const BLOCK_MATH_INDICATOR = '\\[';
export const INLINE_MATH_INDICATOR = '\\(';
export const BLOCK_MATH_REGEX = /(\\\[(.*?)\\])/g;
export const INLINE_MATH_REGEX = /(\\\((.*?)\\\))/g;
export const RADIX = 10;

export const FILTER_ALL_SPACE_ID = 'defaultid';
export const DEFAULT_AUTHENTICATION = false;
export const AUTHENTICATED = 'authenticated';

export const SYNC_ADDED = 'added';
export const SYNC_REMOVED = 'removed';
export const SYNC_MOVED = 'moved';
export const SYNC_BEFORE_MOVED = 'beforeMoved';
export const SYNC_UPDATED = 'updated';

const DIFF_COLORS = {
  [SYNC_ADDED]: '#ddffde',
  [SYNC_REMOVED]: '#fddedb',
  [SYNC_UPDATED]: '#f9edc5',
  [SYNC_MOVED]: 'lightgrey',
};

export const DIFF_STYLES = {
  [SYNC_UPDATED]: {
    background: DIFF_COLORS[SYNC_UPDATED],
  },
  [SYNC_REMOVED]: {
    background: DIFF_COLORS[SYNC_REMOVED],
  },
  [SYNC_ADDED]: {
    background: DIFF_COLORS[SYNC_ADDED],
  },
  [SYNC_MOVED]: {
    background: DIFF_COLORS[SYNC_MOVED],
  },
  [SYNC_BEFORE_MOVED]: {
    background: DIFF_COLORS[SYNC_MOVED],
    opacity: 0.5,
  },
};

export const DEFAULT_SYNC_ADVANCED_MODE = false;
export const SYNC_SPACE_PROPERTIES = [
  'name',
  'image',
  'description',
  'phases',
  'items',
];
export const SYNC_ITEM_PROPERTIES = ['name', 'description', 'content'];
