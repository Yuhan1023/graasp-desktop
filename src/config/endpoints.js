const { REACT_APP_GRAASP_HOST } = process.env;

// eslint-disable-next-line import/prefer-default-export
export const generateGetSpaceEndpoint = id =>
  `${REACT_APP_GRAASP_HOST}/spaces/${id}/download-offline`;
