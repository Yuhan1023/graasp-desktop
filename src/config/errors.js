
// same constants exist in electron.js make sure you change both
export const ERROR_ZIP_CORRUPTED = 'ERROR_ZIP_CORRUPTED';
export const ERROR_JSON_CORRUPTED = 'ERROR_JSON_CORRUPTED';
export const ERROR_SPACE_ALREADY_AVAILABLE = 'ERROR_SPACE_ALREADY_AVAILABLE';
export const ERROR_GENERAL = 'ERROR_GENERAL';
