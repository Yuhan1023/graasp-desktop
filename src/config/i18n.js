import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';
import en from '../langs/en';
import fr from '../langs/fr';

i18n.use(initReactI18next).init({
  resources: {
    en_ALL: en,
    fr_ALL: fr,
  },
  fallbackLng: 'en_ALL',
  // debug only when not in production
  debug: process.env.NODE_ENV !== 'production',
  ns: ['translations'],
  defaultNS: 'translations',
  keySeparator: false,
  interpolation: {
    escapeValue: false,
    formatSeparator: ',',
  },
  react: {
    wait: true,
  },
});

const langs = {
  // bg_ALL: "български",
  // ca_ALL: "Català",
  // cs_ALL: "čeština",
  // de_ALL: "Deutsch",
  // el_ALL: "Ελληνικά",
  en_ALL: 'English',
  // es_ALL: "Español",
  // et_ALL: "Eesti",
  // fi_ALL: "Suomi",
  fr_ALL: 'Français',
  // hu_ALL: "Magyar",
  // it_ALL: "Italiano",
  // ja_ALL: "日本語",
  // ka_ALL: "ქართული",
  // lt_ALL: "lietuvių kalba",
  // lv_ALL: "Latviešu",
  // nl_ALL: "Nederlands",
  // pt_ALL: "Português",
  // ro_ALL: "Română",
  // ru_ALL: "Русский",
  // sk_ALL: "Slovenský",
  // sl_ALL: "Slovenščina",
  // sr_ALL: "српски језик",
  // sw_ALL: "Kiswahili",
  // tr_ALL: "Türkçe",
  // uk_ALL: "Українська",
  // vi_ALL: "Tiếng Việt",
  // zh_ALL: "简体中文",
  // zh_tw: "繁體中文",
};

export { langs };

export default i18n;
