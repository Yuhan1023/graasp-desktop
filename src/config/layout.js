export const MAX_APP_HEIGHT = 600;
export const MIN_APP_HEIGHT = 50;
export const DEFAULT_APP_HEIGHT = 300;
