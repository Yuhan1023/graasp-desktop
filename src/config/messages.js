// this file needs to use module.exports as it is used both by react and electron

const ERROR_DOWNLOADING_MESSAGE = 'There was a problem downloading your files';
const ERROR_LOADING_MESSAGE = 'There was a problem loading your space';
const ERROR_EXPORTING_MESSAGE = 'There was a problem exporting this space';
const SUCCESS_EXPORTING_MESSAGE = 'Space was exported successfully';
const ERROR_DELETING_MESSAGE = 'There was a problem deleting this space';
const SUCCESS_DELETING_MESSAGE = 'Space was deleted successfully';
const ERROR_ZIP_CORRUPTED_MESSAGE =
  'The archive provided is not formatted properly';
const ERROR_JSON_CORRUPTED_MESSAGE = "Space's JSON file is corrupted";
const ERROR_SPACE_ALREADY_AVAILABLE_MESSAGE =
  'A space with the same id is already available';
const SUCCESS_SPACE_LOADED_MESSAGE = 'Space was loaded successfully';
const UNEXPECTED_ERROR_MESSAGE = 'An unexpected error has occurred';
const ERROR_GETTING_SPACE_MESSAGE = 'There was an error getting that space.';
const OFFLINE_ERROR_MESSAGE = 'This functionality requires online connectivity';
const ERROR_MESSAGE_HEADER = 'Error';
const ERROR_SAVING_SPACE_MESSAGE = 'There was an error saving the space';
const SUCCESS_SAVING_MESSAGE = 'Space was saved successfully';
const SUCCESS_MESSAGE_HEADER = 'Success';

module.exports = {
  ERROR_DOWNLOADING_MESSAGE,
  ERROR_MESSAGE_HEADER,
  ERROR_SAVING_SPACE_MESSAGE,
  SUCCESS_SAVING_MESSAGE,
  SUCCESS_MESSAGE_HEADER,
  ERROR_GETTING_SPACE_MESSAGE,
  ERROR_EXPORTING_MESSAGE,
  SUCCESS_EXPORTING_MESSAGE,
  ERROR_DELETING_MESSAGE,
  SUCCESS_DELETING_MESSAGE,
  ERROR_ZIP_CORRUPTED_MESSAGE,
  ERROR_JSON_CORRUPTED_MESSAGE,
  ERROR_SPACE_ALREADY_AVAILABLE_MESSAGE,
  SUCCESS_SPACE_LOADED_MESSAGE,
  UNEXPECTED_ERROR_MESSAGE,
  OFFLINE_ERROR_MESSAGE,
  ERROR_LOADING_MESSAGE,
};
