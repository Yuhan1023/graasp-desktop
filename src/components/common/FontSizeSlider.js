import React from 'react';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/core/styles';
import Grid from '@material-ui/core/Grid';
import { useTranslation } from 'react-i18next';
import Typography from '@material-ui/core/Typography';
import Slider from '@material-ui/core/Slider';
import { connect } from 'react-redux';
import Input from '@material-ui/core/Input';
import FormatSizeIcon from '@material-ui/icons/FormatSize';
import { setFontSize } from '../../actions';
import {
  FONT_SIZE_MAX_VALUE,
  DEFAULT_FONT_SIZE,
  FONT_SIZE_MIN_VALUE,
  FONT_SIZE_STEP,
} from '../../config/constants';
import { FONT_SIZE_SETTING_LABEL_ID } from '../../config/selectors';

const useStyles = makeStyles(() => ({
  root: {
    width: 250,
  },
  input: {
    width: 42,
  },
}));

export const FontSizeSlider = ({ fontSize, dispatchSetFontSize }) => {
  const classes = useStyles();
  const { t } = useTranslation();

  const handleSliderChange = (event, newValue) => {
    dispatchSetFontSize(newValue);
  };

  const handleInputChange = (event) => {
    dispatchSetFontSize(
      event.target.value === '' ? DEFAULT_FONT_SIZE : Number(event.target.value)
    );
  };

  const handleBlur = () => {
    if (fontSize < FONT_SIZE_MIN_VALUE) {
      dispatchSetFontSize(FONT_SIZE_MIN_VALUE);
    } else if (fontSize > FONT_SIZE_MAX_VALUE) {
      dispatchSetFontSize(FONT_SIZE_MAX_VALUE);
    }
  };

  return (
    <div className={classes.root}>
      <Typography id={FONT_SIZE_SETTING_LABEL_ID} gutterBottom>
        {t('Font Size')}
      </Typography>
      <Grid container spacing={2} alignItems="center">
        <Grid item>
          <FormatSizeIcon />
        </Grid>
        <Grid item xs>
          <Slider
            value={fontSize}
            onChange={handleSliderChange}
            aria-labelledby={FONT_SIZE_SETTING_LABEL_ID}
            min={FONT_SIZE_MIN_VALUE}
            max={FONT_SIZE_MAX_VALUE}
            marks
            step={FONT_SIZE_STEP}
          />
        </Grid>
        <Grid item>
          <Input
            className={classes.input}
            value={fontSize}
            margin="dense"
            onChange={handleInputChange}
            onBlur={handleBlur}
            inputProps={{
              min: FONT_SIZE_MIN_VALUE,
              max: FONT_SIZE_MAX_VALUE,
              type: 'number',
              'aria-labelledby': FONT_SIZE_SETTING_LABEL_ID,
            }}
          />
        </Grid>
      </Grid>
    </div>
  );
};

FontSizeSlider.propTypes = {
  dispatchSetFontSize: PropTypes.func.isRequired,
  fontSize: PropTypes.number,
};

FontSizeSlider.defaultProps = {
  fontSize: DEFAULT_FONT_SIZE,
};

const mapStateToProps = ({ authentication }) => ({
  fontSize: authentication.getIn(['user', 'settings', 'fontSize']),
});

const mapDispatchToProps = {
  dispatchSetFontSize: setFontSize,
};

const ConnectedComponent = connect(
  mapStateToProps,
  mapDispatchToProps
)(FontSizeSlider);

export default ConnectedComponent;
