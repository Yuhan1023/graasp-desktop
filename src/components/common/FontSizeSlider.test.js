import React from 'react';
import { shallow } from 'enzyme';
import Typography from '@material-ui/core/Typography';
import Slider from '@material-ui/core/Slider';
import Input from '@material-ui/core/Input';
import { FontSizeSlider } from './FontSizeSlider';
import { DEFAULT_FONT_SIZE, FONT_SIZE_MIN_VALUE } from '../../config/constants';

const createFontSizeSliderProps = ({ fontSize } = {}) => ({
  dispatchSetFontSize: jest.fn(),
  fontSize,
});

describe('<FontSizeSlider />', () => {
  let wrapper;

  describe('default render', () => {
    beforeAll(() => {
      const props = createFontSizeSliderProps();
      // eslint-disable-next-line react/jsx-props-no-spreading
      wrapper = shallow(<FontSizeSlider {...props} />);
    });

    it('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('renders one <Slider /> and one <Input /> component with default font size', () => {
      const text = wrapper.find(Typography);
      expect(text).toHaveLength(1);
      expect(text.contains('Font Size')).toBeTruthy();

      const slider = wrapper.find(Slider);
      expect(slider.prop('value')).toBe(DEFAULT_FONT_SIZE);

      const input = wrapper.find(Input);
      expect(input.prop('value')).toBe(DEFAULT_FONT_SIZE);
    });
  });

  describe('correct font size render', () => {
    beforeAll(() => {
      const props = createFontSizeSliderProps({
        fontSize: FONT_SIZE_MIN_VALUE,
      });
      // eslint-disable-next-line react/jsx-props-no-spreading
      wrapper = shallow(<FontSizeSlider {...props} />);
    });

    it('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('renders one <Slider /> and one <Input /> component with given font size', () => {
      const text = wrapper.find(Typography);
      expect(text).toHaveLength(1);
      expect(text.contains('Font Size')).toBeTruthy();

      const slider = wrapper.find(Slider);
      expect(slider.prop('value')).toBe(FONT_SIZE_MIN_VALUE);

      const input = wrapper.find(Input);
      expect(input.prop('value')).toBe(FONT_SIZE_MIN_VALUE);
    });
  });
});
