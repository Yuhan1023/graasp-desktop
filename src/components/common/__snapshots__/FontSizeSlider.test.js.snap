// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FontSizeSlider /> correct font size render renders correctly 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(Typography))
    gutterBottom={true}
    id="fontSizeSettingLabel"
  >
    Font Size
  </WithStyles(ForwardRef(Typography))>
  <WithStyles(ForwardRef(Grid))
    alignItems="center"
    container={true}
    spacing={2}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
    >
      <Memo(ForwardRef(FormatSizeIcon)) />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={true}
    >
      <WithStyles(ForwardRef(Slider))
        aria-labelledby="fontSizeSettingLabel"
        marks={true}
        max={20}
        min={10}
        onChange={[Function]}
        step={2}
        value={10}
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
    >
      <WithStyles(ForwardRef(Input))
        className="makeStyles-input-2"
        inputProps={
          Object {
            "aria-labelledby": "fontSizeSettingLabel",
            "max": 20,
            "min": 10,
            "type": "number",
          }
        }
        margin="dense"
        onBlur={[Function]}
        onChange={[Function]}
        value={10}
      />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</div>
`;

exports[`<FontSizeSlider /> default render renders correctly 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(Typography))
    gutterBottom={true}
    id="fontSizeSettingLabel"
  >
    Font Size
  </WithStyles(ForwardRef(Typography))>
  <WithStyles(ForwardRef(Grid))
    alignItems="center"
    container={true}
    spacing={2}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
    >
      <Memo(ForwardRef(FormatSizeIcon)) />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={true}
    >
      <WithStyles(ForwardRef(Slider))
        aria-labelledby="fontSizeSettingLabel"
        marks={true}
        max={20}
        min={10}
        onChange={[Function]}
        step={2}
        value={14}
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
    >
      <WithStyles(ForwardRef(Input))
        className="makeStyles-input-2"
        inputProps={
          Object {
            "aria-labelledby": "fontSizeSettingLabel",
            "max": 20,
            "min": 10,
            "type": "number",
          }
        }
        margin="dense"
        onBlur={[Function]}
        onChange={[Function]}
        value={14}
      />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</div>
`;
